openshiftcontrolplane/v1/types.go

```diff
@@ -10,6 +10,7 @@ import (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// +openshift:compatibility-gen:level=1
```

Comment:
on-disk configuration is getting weird.  

I think we need an option for `openshift:compatibility-gen:exposed=false`, which can be paired with `openshift:compatibility-gen:level=1` for our on-disk configuration types that we don't allow customers to set.